### ISSUE TITLE
Option to use haptic feedback as controller rumble

### ISSUE THREAD

**Issue by johnz133** [moonlight-stream/moonlight-ios#445]
Is it possible to hijack the controller rumble and use the phone's haptic feedback for vibration instead?

**Comment by EmilyLove26** [comment]
Yes please! I use the Razer Kishi which does not have built-in rumble. It would be nice to redirect rumble to haptic feedback.

**Comment by EmilyLove26** [comment]
Bump. Still a desired feature!!

**Comment by codevski** [comment]
Bump, the back bone one is a great controller would love to use phones built in rumble

**Comment by ZaydelSenpai** [comment]
I want this so bad. Hope!

**Comment by SalmonCactus763** [comment]
bump, would make games better

**Comment by codevski** [comment]
So the backbone app has rumble and it feels like it comes from the control when it isn't. I wonder if it's possible to translate rumble that moonlight would of sent to the phones rumble that would be awesome

**Comment by almanzarsaul** [comment]
+1, would love to see this as an option as well.

**Comment by trevlars** [comment]
Please add! This would complete my gamesir iPhone setup!

**Comment by ZaydelSenpai** [comment]
I want this so bad. Hope!

**Comment by laralove143** [comment]
please add this pleasee

**Comment by trevlars** [comment]
I would absolutely love this with a GameSir! Tempted to use joycon now just to get vibration

**Comment by codevski** [comment]
might be a limitation to iOS? feature seems to be implemented on android using the gamesir and rumble coming from my pixel without any issues. Settings are in moonlight so it should be supported since the backbone one app has rumble support when navigating menus

**Comment by laralove143** [comment]
yeah it's just not implemented on ios :(

**Comment by balu021081** [comment]
Any news on that feature request? Thanks for this great app btw.

**Comment by nevespt** [comment]
> might be a limitation to iOS? feature seems to be implemented on android using the gamesir and rumble coming from my pixel without any issues. Settings are in moonlight so it should be supported since the backbone one app has rumble support when navigating menus

It isnt, steam link uses phone as rumble even using backbone v2

**Comment by life1st** [comment]
https://www.youtube.com/watch?v=C2KNX90tibM
I found a video, looks like he make a custom version of moonlight, and support vibrate, only changed 2 lines of code.

**Comment by laralove143** [comment]
> https://www.youtube.com/watch?v=C2KNX90tibM I found a video, looks like he make a custom version of moonlight, and support vibrate, only changed 2 lines of code.

i transcribed and translated the video with whisper, results may vary but it gets the idea across

Hello everyone, I believe many people like me On the iOS platform, I especially hope that my controller can support vibration and current flow And then there's the G8 controller It doesn't have an automatic motor And then this one on the Android platform has a Let the device simulate vibration And then the iOS platform has never had one And then today I modified its code again And then normally in this control center the two Joy-Con handles in the control center are vibrated. Then go to the Apple document on the third EPP interface. This Joy-Con does not support vibration. I thought about changing it to the same as Android. This only supports iPhone above iOS 13 Because only iPhone has a vibration motor Then show you the effect These are two Joy-Con Normally it is not vibrating Now listen to everyone In this case, the phone is vibrating and the Joy-Con is not vibrating If you hold it in your hand with this kind of stretcher It feels like the handle is vibrating, which is not bad Although the original God is on the XBOX handle Vibration is not as much as PS5 But there are still a few, such as this explosive barrel Finally, let's share the modification One file, two codes It was originally a direct return Now I give it a new A mobile phone vibration Then this video is here Thank you for watching If the video is helpful to everyone Welcome to like and subscribe See you next time Bye bye
